### PR TITLE
chore(output): add output.module and output.iife

### PR DESCRIPTION
**Pull request by rishabh3112**
_describe your changes..._
Part of https://github.com/webpack/webpack.js.org/issues/3312

### CODE REVIEW

File: src/content/configuration/output.md [hunk 1]
```diff
@@ -1088,4 +1088,36 @@ module.exports = {
...
 };
-```
\ No newline at end of file
+```
+
+## `output.iife`
+
+`boolean = true`
+
+Tells webpack to add iife wrapper around emitted code
```
Comment (montogeek) [inline src/content/configuration/output.md]:
```suggestion
Tells webpack to add [IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE) wrapper around emitted code.
```

File: src/content/configuration/output.md [hunk 2]
```diff
@@ -1088,4 +1088,36 @@ module.exports = {
...
+    iife: true
+  }
+};
+```
+
+## `output.module`
+
+`boolean = true`
+
+Allow outputting javascript files as module type. It set  `output.iife` to `false`, `output.libraryTarget` to `"module"`, `output.jsonpScriptType` to `"module"` and `terserOptions.module` = `true`
```
Comment (montogeek) [inline src/content/configuration/output.md]:
```suggestion
Allow outputting JavaScript files as module type. It set `output.iife` to `false`, `output.libraryTarget` to `"module"`, `output.jsonpScriptType` to `"module"` and `terserOptions.module` = `true`
```

Comment (jamesgeorge007) [inline src/content/configuration/output.md]:
```suggestion
Allow outputting javascript files as module type. It sets  `output.iife` to `false`, `output.libraryTarget` to `"module"`, `output.jsonpScriptType` to `"module"` and `terserOptions.module` = `true`
```

File: src/content/configuration/output.md [hunk 3]
```diff
@@ -1088,4 +1088,36 @@ module.exports = {
...
+    iife: true
+  }
+};
+```
+
+## `output.module`
+
+`boolean = true`
+
+Allow outputting JavaScript files as module type. It set `output.iife` to `false`, `output.libraryTarget` to `"module"`, `output.jsonpScriptType` to `"module"` and `terserOptions.module` = `true`
```
Comment (rishabh3112) [inline src/content/configuration/output.md]:
```suggestion
Allow outputting JavaScript files as module type. It sets `output.iife` to `false`, `output.libraryTarget` to `"module"`, `output.jsonpScriptType` to `"module"` and `terserOptions.module` to `true`
```

File: src/content/configuration/output.md (hunk 2 above)
Comment (rishabh3112) [reply src/content/configuration/output.md]: I wasn't allowed to apply this suggestion after applying suggestion from @montogeek on the same line. I have tagged you into the commit instead.

**Comment by montogeek** [comment]
Thanks!

**Pull request merged**